src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1060,10 +1082,9 @@ public void EvictQueries(string cacheRegion)
 			{
 				if (settings.IsQueryCacheEnabled)
 				{
-					IQueryCache currentQueryCache;
-					if (queryCaches.TryGetValue(cacheRegion, out currentQueryCache))
+					if (queryCaches.TryGetValue(cacheRegion, out var currentQueryCache))
 					{
-						currentQueryCache.Clear();
+						currentQueryCache.Value.Clear();
```

Comment:
I think, here we might need to check if the cache was actually created?

Reply:
Here this is a `TryGetValue`, so it will not create the `Lazy`. If there is one, it means it has been created by some other function calls, which will forcibly also create the lazy value. A concurrency scenario may still causes the call here to create the lazy value before the other call (the one having created the lazy), but that is not an issue, this other call would have created the value itself anyway.